### PR TITLE
Message

### PR DESCRIPTION
**Pull request by Hepic**

### CODE REVIEW

File: client/message/history.jsx [hunk 1]
```diff
@@ -31,6 +31,15 @@ const History = React.createClass({
             myUsername: null
         };
     },
+    deleteLiveMessage(username) {
+        var messages = this.state.messages;
+
+        for(var messageid in messages) {
+            if (messages[messageid].username == username) {
+                delete messages[messageid];
```
Comment (dionyziz) [inline client/message/history.jsx]: You need to setState to let react know you're modifying the state.

File: realtime/server.js [hunk 2]
```diff
@@ -192,7 +192,7 @@ socket.on('connection', function (client) {
         }
 
         messages_typing = messagesTypingNew;
-
+
```
Comment (dionyziz) [inline realtime/server.js]: Remove trailing whitespace

File: client/message/history.jsx [hunk 3]
```diff
@@ -31,6 +31,17 @@ const History = React.createClass({
             myUsername: null
         };
     },
+    deleteLiveMessage(username) {
+        var messages = this.state.messages;
+
+        for(var messageid in messages) {
```
Comment (gtklocker) [inline client/message/history.jsx]: nit

File: etc/spec/SPECIFICATION.md [hunk 4]
```diff
@@ -449,7 +449,8 @@ works as follows.
 Messages in ting are separated into persistent and non-persistent. Persistent
 messages are messages that have been sent. Non-persistent messages are messages
 that are currently being typed, but have not been sent yet (the Enter key has
-not been pressed by the user.) Users can see both persistent and non-persistent
+not been pressed by the user.). Non-persistent message of a user that disconnects
```
Comment (dionyziz) [inline etc/spec/SPECIFICATION.md]: "A non-persistent message"

File: client/message/history.jsx [hunk 5]
```diff
@@ -31,6 +31,17 @@ const History = React.createClass({
             myUsername: null
         };
     },
+    deleteLiveMessage(username) {
```
Comment (dionyziz) [inline client/message/history.jsx]: We don't use the term "live" anywhere else. Maybe we can name this "deleteTypingMessage".

**Comment by dionyziz** [comment]
Squash

File: client/ting.jsx [hunk 6]
```diff
@@ -155,11 +155,11 @@ const Ting = React.createClass({
...
 i18n.init(
-    { 
+    {
         resGetPath:' locales/__lng__.json',
         lng: 'el-GR'
     },
     () => {
         React.render(<Ting />, document.getElementsByClassName('ting')[0]);
-    }   
+    }
```
Comment (dionyziz) [inline client/ting.jsx]: Move these stylistic changes to separate commit or remove completely.

**Comment by dionyziz** [comment]
Limit your commit messages to 70 characters.

**Comment by dionyziz** [comment]
Order of commits should be: Spec first, then implementation

**Comment by dionyziz** [comment]
Specification commit message is confusing. The spec isn't "informed". Perhaps use "Modify"?

File: etc/spec/SPECIFICATION.md [hunk 7]
```diff
@@ -449,7 +449,8 @@ works as follows.
 Messages in ting are separated into persistent and non-persistent. Persistent
 messages are messages that have been sent. Non-persistent messages are messages
 that are currently being typed, but have not been sent yet (the Enter key has
-not been pressed by the user.) Users can see both persistent and non-persistent
+not been pressed by the user.). A non-persistent message of a user that disconnects
```
Comment (dionyziz) [inline etc/spec/SPECIFICATION.md]: Limit lines to 80 characters

File: client/message/history.jsx [hunk 8]
```diff
@@ -31,6 +31,17 @@ const History = React.createClass({
...
         };
     },
+    deleteTypingMessage(username) {
+        var messages = this.state.messages;
+
+        for (var messageid in messages) {
+            if (messages[messageid].username == username) {
+                delete messages[messageid];
+            }
+        }
```
Comment (dionyziz) [inline client/message/history.jsx]: This for loop can be written in one line using [_.remove](https://lodash.com/docs#remove).
